specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2023-03-01-preview/amlfilesystem.json

```diff
@@ -0,0 +1,1121 @@
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "storagecache.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "amlFilesystems_Get": {
+            "$ref": "./examples/amlFilesystems_Get.json"
+          }
+        },
+        "tags": [
+          "amlFilesystems"
+        ],
+        "description": "Returns an AML file system.",
+        "operationId": "amlFilesystems_Get"
+      },
+      "put": {
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/AmlFilesystemNameParameter"
+          },
+          {
+            "description": "Object containing the user-selectable properties of the AML file system. If read-only properties are included, they must match the existing values of those properties.",
+            "in": "body",
+            "name": "amlFilesystem",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AmlFilesystem"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The AML file system has been created or updated.",
+            "schema": {
+              "$ref": "#/definitions/AmlFilesystem"
+            }
+          },
+          "201": {
+            "description": "AML file system creation has been initiated. Poll the AML file system's provisioningState property to monitor creation progress.",
+            "headers": {
+              "azure-async-operation": {
+                "description": "URI to poll for the operation status",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/AmlFilesystem"
+            }
+          },
+          "202": {
+            "description": "AML file system creation or update has been initiated. Poll the AML file system's provisioningState property to monitor creation/update progress.",
+            "headers": {
+              "Location": {
+                "description": "Location URI to poll for result",
+                "type": "string"
+              },
+              "azure-async-operation": {
+                "description": "URI to poll for the operation status",
+                "type": "string"
+              }
+            }
+          },
```

Comment:
You're getting lintdiff here 

❌ CreateOperationAsyncResponseValidation Only 201 is the supported response code for PUT async response.
Location: Microsoft.StorageCache/preview/2023-03-01-preview/amlfilesystem.json#L278 RPC-Async-V1-01

because the recommended pattern for async PUT requets is 201 and ProvisioningState...

See [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

Reply:
> You're getting lintdiff here
> 
> ❌ CreateOperationAsyncResponseValidation Only 201 is the supported response code for PUT async response. Location: Microsoft.StorageCache/preview/2023-03-01-preview/amlfilesystem.json#L278 RPC-Async-V1-01
> 
> because the recommended pattern for async PUT requets is 201 and ProvisioningState...
> 
> See https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put

Our PUT returns 202 in the modify/update case.  This works the same way the storagecache does.  It allows modifications of the fields that can change or the same content for idempotent retries.

Reply:
Do you have a suggestion for this error in the tool or the spec?  We support patch for partial modifies and PUT for modifies containing the full body.  When the object is already existing should we just return 201 all the time when it is async because it is a PUT and not a PATCH?

I can make the update in the code or swagger either way or seek an exception in the lintdiff.
Just want to make sure I make the right fix.

Thanks.

Reply:
PUT means Create or Update... In line with HTTP, when the object is already existing, you can return 200, to differentiate the updated-in-place case from the created case.

Reply:
I fixed it and pushed the update.  The linter error went away.
I make the code change for that case.
Thanks.